mqtt/mqtt.js

```diff
@@ -87,125 +57,77 @@ nats.subscribe('channel.*', function (msg) {
+                Payload: packet.payload
             });
 
-            console.log(msg);
-
-            console.log(util.inspect(packet, false, null))
-
             // Pub on NATS
-            nats.publish('channel.' + channel, msg);
+            nats.publish('channel.' + channel, rawMsg);
         } else {
-            console.log('Publish not authorized');
-            error = 4; // Bad username or password
+            logger.warn("unauthorized publish: %s", err.message);
+            callback(4); // Bad username or password
         }
-        callback(error);
     });
 };
 
 // AuthZ SUB
 aedes.authorizeSubscribe = function (client, packet, callback) {
-    // Topics are in the form `mainflux/channels/<channel_id>/messages/senml-json`
-    var channel = packet.topic.split('/')[2];
-    /**
-    * Check if PUB is authorized
-    */
-    var options = {
-        url: config.auth_url + ':' + config.auth_port + '/channels/' + channel + '/access-grant',
-        method: 'GET',
-        headers: {
-            'Content-Type': 'application/json',
-            'Authorization': client.password
-        }
-    };
-
-    request(options, function (err, res) {
-        var error = null;
-        if (res && (res.statusCode === 200)) {
-            console.log('Subscribe authorized OK');
+    // Topics are in the form `channels/<channel_id>/messages/senml-json`
+    var channel = packet.topic.split('/')[1];
+    
+    things.canAccess({
+        token: client.password,
+        chanID: channel
+    }, function (err, res) {
+        if (!err) {
+            logger.info('authorized subscribe');
+            callback(null, packet);
         } else {
-            console.log('Subscribe not authorized');
-            error = 4; // Bad username or password
+            logger.warn('unauthorizerd subscribe: %s', err);
+            callback(4, packet); // Bad username or password
         }
-        callback(error, packet);
     });
 };
 
 // AuthX
 aedes.authenticate = function (client, username, password, callback) {
-    var c = client;
-    var options = {
-        url: config.auth_url + ':' + config.auth_port + '/access-grant',
-        method: 'GET',
-        headers: {
-            'Content-Type': 'application/json',
-            'Authorization': password
-        }
-    };
-    request(options, function (err, res) {
-        var error = null;
-        var success = null;
-        if (res && (res.statusCode === 200)) {
-            // Set MQTT client.id to correspond to Mainflux device UUID
-            c.id = res.headers['x-client-id'];
-            // Store password for future references
-            c.password = password;
-            success = true;
-        } else {
-            error = new Error('Auth error');
-            error.returnCode = 4; // Bad username or password
-            success = false;
-        }
-        // Respond with auth error and success
-        callback(error, success);
-    });
```

Comment:
Without authentication check you are letting any client connect, i.e. it will send `CONNECT` MQTT req, and servier will respond with `CONNACK` and TCP connection will be established, even though client did not do any SUB or PUB yet.